docs/cip/README.md

```diff
@@ -4,518 +4,1132 @@ Title: Ouroboros Leios - Greater transaction throughput
+<a id="service-quality-differentiation" href="#service-quality-differentiation"></a>**Service Quality Differentiation**: The inherent service difference between RB and EB inclusion (immediate vs. certificate-dependent confirmation) creates natural economic incentives without requiring protocol-level price differentiation. Applications requiring immediate confirmation can adjust their fee strategies accordingly within the existing framework.
 
-### Metrics
+**Operational Cost Optimization**
 
-> [!NOTE]
-> 
-> This is a preliminary set of metrics that will be finalized when the Leios protocol variants are finalized and the simulation studies are complete.
+<a id="adaptive-eb-production" href="#adaptive-eb-production"></a>**Adaptive EB Production**: To address concerns about operational costs during low-traffic periods, the protocol implements adaptive EB announcement based on transaction availability. SPOs should only announce EBs when they contain sufficient transactions (e.g., 10% capacity utilization) to justify the voting overhead costs. When traffic levels can be adequately served by RBs alone, no EBs are announced, reducing operational costs to Praos levels.
+
+<a id="cost-scaling-mechanism" href="#cost-scaling-mechanism"></a>This adaptive mechanism ensures the protocol's cost structure scales with actual throughput demand rather than imposing fixed overhead regardless of network utilization.
+
+<a id="hardware-upgrade-incentives" href="#hardware-upgrade-incentives"></a>**Hardware Upgrade Incentives**: The increased computational and bandwidth requirements for Leios operation are offset by higher potential rewards from increased transaction throughput. As demonstrated in the [operating costs analysis](#operating-costs), SPO profitability improves significantly once sustained throughput exceeds 30-50 TPS, providing clear economic incentives for infrastructure upgrades.
+
+**Reserve Contribution Adjustments**
+
+<a id="reserve-contribution-adjustments" href="#reserve-contribution-adjustments"></a>During low-traffic periods when EB production is minimal, the protocol maintains current Reserve contribution patterns without artificial inflation. The adaptive production mechanism ensures that enhanced infrastructure costs are only incurred when corresponding revenue opportunities exist through increased transaction processing.
+
+## Rationale
+
+### How Leios addresses CPS-18 and increases throughput
+
+The [Leios research paper][leios-paper] describes a highly concurrent protocol with three block types - Input Blocks (IBs), Endorser Blocks (EBs), and Ranking Blocks (RBs)-produced independently across decoupled, pipelined stages. This specification simplifies that design by eliminating IBs and coupling EB production with RB production, reducing complexity while preserving substantial throughput gains.
+
```

Comment:
Would be great to have summarizing statement of the idea behind Linear Leios, something like: Linear Leios introduces a voting layer over Nakamoto-style consensus for dealing with transaction surplus.